sdk/core/Azure.Core.TestFramework/src/IgnoreServiceErrorAttribute.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Globalization;
+
+namespace Azure.Core.TestFramework
+{
+    /// <summary>
+    /// Marks tests failing with specific service error codes as "inconclusive".
+    /// </summary>
+    /// <remarks>
+    /// This is intended for use with intermittent service issues that happen only during internal testing,
+    /// or that are temporary issues causing test failures but will be fixed. Please engage with service teams
+    /// before using this attribute to see if live test issues can be resolved first.
+    /// </remarks>
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = true, Inherited = true)]
+    public class IgnoreServiceErrorAttribute : Attribute
```

Comment:
Should probably document that this only applies to RecordedTest tests. If possible, we should throw if used on a test without RecordedTest.

Reply:
I'll document it, but it's (mostly: I did add a simple match method) just a data-only attribute. You can't really control to what it applies anyway, though there's thing I could do to make sure if it gets used it's base is RecordedTestBase, but not sure that's needed right now.

Reply:
How this?

```     /// <summary>
    /// Marks tests failing with specific service error codes as "inconclusive". Must be used with test fixtures derived
    /// from <see cref="RecordedTestBase"/> and is used only by tests attributed with <see cref="RecordedTestAttribute"/>.
    /// </summary>
```

Reply:
I would think you could check it in RecordedTestBase Setup method and if RecordedTest is not there, then throw.
We do something similar in ClientTestBase - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core.TestFramework/src/ClientTestBase.cs#L58-L59

Reply:
> How this?
> 
> ```
>     /// Marks tests failing with specific service error codes as "inconclusive". Must be used with test fixtures derived
>     /// from <see cref="RecordedTestBase"/> and is used only by tests attributed with <see cref="RecordedTestAttribute"/>.
>     /// </summary>
> ```

That works.